enhancements/network/persistent-ips-sdn-secondary-networks.md

```diff
@@ -403,7 +403,46 @@ TODO
 
 ### Integrate the `PersistentIP` CRD with OVN-Kubernetes ip pool
 
-TODO
+We plan on integrating the proposed `PersistentIP`s CRD with the existing
+OVN-Kubernetes IP pool allocation mechanism by reacting to two different
+events: ovnkube-master start, and `PersistentIP` CR delete.
+
+The updates to the `PersistentIP` CR will not be supported, and since they will
+be provisioned as part of the regular pod triggered CNI ADD operation, it is
+not necessary to react to provisioned `PersistentIP`s.
+
+#### How to integrate the new CRD with the OVN-K IP pool on ovnkube-master start
+
+When the ovnkube-master starts, it will - for every pod in the system - invoke
+the `AllocateIPs` function from the allocator. We will need to also read all
+the `PersistentIP`s from the informer cache, and invoke the `AllocateIPs`
+function for each of the persisted IPs, as depicted in the following diagram.
+
+```mermaid
+sequenceDiagram
+
+title OVN-Kubernetes cache filling (on controller startup)
+
+loop all persistent IPs
+  Kubernetes apiserver->>ovn-k controller: PersistentIPs
+  ovn-k controller->>ovn-k controller: Cache PersistentIPs at informer
+end
+
+ovn-k controller->>ovn-k ippool: AllocateIPs(persistent IPs)
+```
+
+For running VMs, it is expected the allocation to fail when provisioning the IP
+(since it was already allocated for the running pod). In this case, we simply
+ignore the error and return.
```

Comment:
You mean calling allocate twice at the same IP ?

Reply:
yes. It's what I currently see in the OVN code; [for instance](https://github.com/ovn-org/ovn-kubernetes/blob/ed09e945f5a655026776844b3a6640a31c295a8a/go-controller/pkg/ovn/base_network_controller_pods.go#L807).